src/coredump/coredumpctl.c

```diff
@@ -883,15 +907,30 @@ static int dump_core(int argc, char **argv, void *userdata) {
         return 0;
 }
 
-static int run_gdb(int argc, char **argv, void *userdata) {
+static int run_debug(int argc, char **argv, void *userdata) {
         _cleanup_(sd_journal_closep) sd_journal *j = NULL;
-        _cleanup_free_ char *exe = NULL, *path = NULL;
+        _cleanup_free_ char *exe = NULL, *path = NULL, *fork_name;
         bool unlink_path = false;
         const char *data;
         size_t len;
         pid_t pid;
         int r;
 
+        if (streq(argv[0], "gdb")) {
+                arg_debugger = "gdb";
+        } else if (arg_debugger == NULL) {
+                char *env_debugger;
+
+                env_debugger = getenv("SYSTEMD_DEBUGGER");
+                if (env_debugger != NULL) {
```

Comment:
we usually rely on c's downgrade-to-bool feature for pointer checks, i.e. shorten `if (e != NULL)` to `if (e)`